src/lib/Synths.ts

```diff
@@ -54,121 +55,127 @@ class Synths {
   }
 
   async getLSPPortfolio() {
-    try {
-      const portfolio: { [key: string]: Record<string, unknown> } = {};
-
-      for (const assetCycles in this.assets) {
-        for (const asset of this.assets[assetCycles]) {
-          if (isAssetConfigLSP(asset)) {
-            const userAddress = await this.#signer.getAddress();
-            const contract = LongShortPairEthers__factory.connect(
-              asset.lsp.address,
-              this.#multicallProvider
-            );
-            const call = prepareLSPStateCall(contract);
-            const [
-              expirationTimestamp,
-              collateralToken,
-              priceIdentifier,
-              pairName,
-              longToken,
-              shortToken,
-              collateralPerPair,
-              timerAddress,
-            ] = await call;
-
-            console.log(
-              priceIdentifier,
-              pairName,
-              collateralPerPair,
-              timerAddress
-            );
-
-            const collateralContract = ERC20Ethers__factory.connect(
-              collateralToken,
-              this.#multicallProvider
-            );
-            const longTokenContract = ERC20Ethers__factory.connect(
-              longToken,
-              this.#multicallProvider
-            );
-            const shortTokenContract = ERC20Ethers__factory.connect(
-              shortToken,
-              this.#multicallProvider
+    const portfolio = [];
+
+    for (const assetCycles in this.assets) {
+      for (const asset of this.assets[assetCycles]) {
+        if (isAssetConfigLSP(asset)) {
+          const userAddress = await this.#signer.getAddress();
+          const contract = LongShortPairEthers__factory.connect(
+            asset.lsp.address,
+            this.#multicallProvider
+          );
+          const call = prepareLSPStateCall(contract);
+          const [
+            expirationTimestamp,
+            collateralToken,
+            ,
+            ,
+            longToken,
+            shortToken,
+            ,
+            ,
+          ] = await call;
```

Comment:
some reason for this?

Reply:
We don't need all values that result from the multicall. As we use this multicall twice we added a helper to prepare the promises.